src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -5204,6 +5269,38 @@ private char[] getPass(String modifier, String arg) {
         tinyHelp();
         return null;    // Useless, tinyHelp() already exits.
     }
+
+    private static class SecretKeyConstraintsParameters implements ConstraintsParameters {
+        private final Key key;
+        private SecretKeyConstraintsParameters(Key key) {
+            this.key = key;
+        }
+
+        @Override
+        public boolean anchorIsJdkCA() {
+            return false;
+        }
+
+        @Override
+        public Set<Key> getKeys() {
+            return (key == null) ? Set.of() : Set.of(key);
```

Comment:
key should never be null, so you don't need to check for this.

Reply:
Removed the extra check.